hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java

```diff
@@ -44,9 +55,16 @@ private boolean validateWithTemplate(String address, ConstraintValidatorContext
         return true;
     }
 
-    protected void createConstraintMessage(ConstraintValidatorContext context, String message) {
+    private void createConstraintMessage(ConstraintValidatorContext context, String message) {
         context.buildConstraintViolationWithTemplate(message)
                 .addConstraintViolation()
                 .disableDefaultConstraintViolation();
     }
+
+    private boolean isInvalidHost(UriTemplate template) {
+        Map<String, Object> uriKeysWithEmptyValues = Arrays.asList(template.getVariables()).stream().collect(toMap(v -> v, v -> "empty"));
+
+        //check if host is null due to bug in jdk http://bugs.java.com/view_bug.do?bug_id=5083594
+        return URI.create(template.expand(uriKeysWithEmptyValues)).getHost() == null;
```

Comment:
I think it's not a good solution to propagate this JDK behaviour into our code. If it doesn't work we should use something different to validate the uri.